src/Navigation/WPF/Navigation.WPF/NavigationServiceExtensions.cs

```diff
@@ -235,19 +234,38 @@ public static class NavigationServiceExtensions
         /// <summary>
         /// Removes the registration associated with the given ViewModel interface type
         /// </summary>
-        /// <typeparam name="TViewModelInterface">ViewModel type to remove from registrations</typeparam>
+        /// <typeparam name="TViewModel">ViewModel type to remove from registrations</typeparam>
         /// <param name="navigationService"></param>
         /// <returns>Navigation service instance</returns>
-        public static INavigationService Unregister<TViewModelInterface>(
+        public static INavigationService Unregister<TViewModel>(
             this INavigationService navigationService)
 
-            where TViewModelInterface : class, IViewModel
+            where TViewModel : class, IViewModel
+        {
+            navigationService.Unregister<TViewModel>();
+            return navigationService;
+        }
+
+        public static INavigationService Unregister<TViewModel, TView>(
+            this INavigationService navigationService)
+
+            where TViewModel : class, IViewModel
+            where TView : FrameworkElement
         {
-            navigationService.Unregister<TViewModelInterface>();
+            navigationService.Unregister<TViewModel, TView>();
             return navigationService;
         }
 
 
+        public static RegistrationInfo WithCondition(
+            this RegistrationInfo registration,
+            Func<bool> condition)
+        {
+            registration.Condition = condition;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this method to add validation of parameter 'registration' before using it.](https://app.codacy.com/gh/UltimateCodeMonkeys/CodeMonkeysMVVM/pullRequest?prid=5575746)